src/ckb/channel.rs

```diff
@@ -2897,6 +2919,87 @@ impl ChannelActorState {
         Ok(())
     }
 
+    fn handle_reestablish_channel_message(
+        &mut self,
+        reestablish_channel: ReestablishChannel,
+        network: &ActorRef<NetworkActorMessage>,
+    ) -> ProcessingChannelResult {
+        match self.state {
+            ChannelState::NegotiatingFunding(_flags) => {
+                // TODO: in current implementation, we don't store the channel when we are in NegotiatingFunding state.
+                // This is an unreachable state for reestablish channel message. we may need to handle this case in the future.
+            }
+            ChannelState::ChannelReady() => {
+                // resend AddTlc, RemoveTlc and CommitmentSigned messages if needed
+                let local_commitment_number = reestablish_channel.local_commitment_number;
+                let mut need_resend_commitment_signed = false;
+                for info in self.tlcs.values() {
```

Comment:
We don't have to repeat all this logic here. https://github.com/nervosnetwork/cfn-node/blob/f9e79b060f7329309eff752b7485bfacc91a629f/src/ckb/channel.rs#L2027 This function may be modified to check whether we need to resend a `AddTlc` or `SendTlc`.